### PR TITLE
Implement formatters for log

### PR DESCRIPTION
**Pull request by bartlomieju**
Based on discussion in #39 I'm making first pass at implementing formatters for log module. 

This is already working and here are examples (they have exactly same output)
1.
```ts
new FileHandler("INFO", { 
    filename: "log.txt", 
    // simple string-based formatter, interpolates with fields of LogRecord
    formatter: "{levelName} {datetime} {msg}" 
});
```
2.
```ts
new FileHandler("INFO", { 
    filename: "log.txt", 
    // function-based formatter, you can do
    // whatever you want here
    formatter: logRecord => {
         return `${logRecord.levelName} {logRecord.datetime} {logRecord.msg}`;
    }
});
```

CC @ry @hayd

### CODE REVIEW

File: log/handlers.ts [hunk 1]
```diff
@@ -2,24 +2,48 @@ import { open, File, Writer } from "deno";
 import { getLevelByName } from "./levels.ts";
 import { LogRecord } from "./logger.ts";
 
+const DEFAULT_FORMATTER = "{levelName} {msg}";
+
+interface HandlerOptions {
+  formatter?: Function | string;
+}
```
Comment (ry) [inline log/handlers.ts]: Can you add a test which usage this?

File: log/handlers.ts [hunk 2]
```diff
@@ -2,24 +2,48 @@ import { open, File, Writer } from "deno";
 import { getLevelByName } from "./levels.ts";
 import { LogRecord } from "./logger.ts";
 
+const DEFAULT_FORMATTER = "{levelName} {msg}";
```
Comment (bartlomieju) [inline log/handlers.ts]: @ry should I set it to glog's format?

File: log/handlers.ts [hunk 3]
```diff
@@ -33,21 +57,27 @@ export class ConsoleHandler extends BaseHandler {
...
 export class FileHandler extends WriterHandler {
   private _file: File;
   private _filename: string;
 
-  constructor(levelName: string, filename: string) {
-    super(levelName);
-    this._filename = filename;
+  constructor(
+    levelName: string, 
+    options: FileHandlerOptions = { filename: "log.txt" }
```
Comment (bartlomieju) [inline log/handlers.ts]: this is just placeholder filename, I want to know your opinion

File: log/handlers.ts (hunk 2 above)
Comment (ry) [reply log/handlers.ts]: No... it was just an idea and it seems people aren't so keen on it. Let's just move forward with this.

File: log/handlers.ts (hunk 3 above)
Comment (ry) [reply log/handlers.ts]: Not so happy with having an options interface with only one item in it. (cc @piscisaureus)
It seems if you're constructing a `FileHandler`, specifying a filename should be part of the process - i.e. no default value.

Comment (bartlomieju) [reply log/handlers.ts]: Actually it's two items, because `FileHandlerOptions` has `filename` and `formatter`. +1 for no default value

**Pull request merged**